### PR TITLE
chore: Enahnce logging around pod failures

### PR DESCRIPTION
**Pull request by simster7**
Example: 
image

Example log:

```
time="2020-10-05T14:02:48-07:00" level=info msg="Pod failed" displayName="retry-container-v2n8p(0)" namespace=argo pod=retry-container-v2n8p-2124340039 templateName=retry-container workflow=retry-container-v2n8p
```

### CODE REVIEW

File: workflow/controller/operator.go [hunk 1]
```diff
@@ -913,7 +913,8 @@ func (woc *wfOperationCtx) podReconciliation() error {
 			// See processNodeRetries for more details.
 			node.FinishedAt = metav1.Time{Time: time.Now().UTC()}
 			woc.wf.Status.Nodes[nodeID] = node
-			woc.log.Warnf("pod %s deleted", nodeID)
+			woc.log.WithField("displayName", node.DisplayName).WithField("templateName", node.TemplateName).
```
Comment (alexec) [inline workflow/controller/operator.go:916]: why not log this is in `markNodePhase`?

Comment (simster7) [reply workflow/controller/operator.go:916]: This is the single point where Pod updates enter the controller. `markNodePhase` may be used for other errors, but the intention of this PR is to enhance the logs related to direct Pod updates

**Comment by sarabala1979** [comment]
Do you need to add the same in `executeTemplate` and all functions which are handling nodes?

**Comment by simster7** [comment]
> Do you need to add the same in executeTemplate and all functions which are handling nodes?

No as stated before this is only relevant for `Pods`, not `Nodes`, and this is the only place where we receive `Pod` updates

**Review by sarabala1979 — APPROVED** [review]
LGTM

File: workflow/controller/operator.go [hunk 2]
```diff
@@ -1052,13 +1053,17 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
...
+				WithField("pod", pod.Name).Infof("Pod failed")
 		}
 		newDaemonStatus = pointer.BoolPtr(false)
 	case apiv1.PodRunning:
 		if pod.DeletionTimestamp != nil {
 			// pod is being terminated
 			newPhase = wfv1.NodeError
 			message = "pod deleted during operation"
+			woc.log.WithField("displayName", node.DisplayName).WithField("templateName", node.TemplateName).
+				WithField("pod", pod.Name).Error(message)
```
Comment (alexec) [inline workflow/controller/operator.go:1066]: should be warning

File: workflow/controller/operator.go [hunk 3]
```diff
@@ -1088,7 +1093,8 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 	default:
 		newPhase = wfv1.NodeError
 		message = fmt.Sprintf("Unexpected pod phase for %s: %s", pod.ObjectMeta.Name, pod.Status.Phase)
-		log.Error(message)
+		woc.log.WithField("displayName", node.DisplayName).WithField("templateName", node.TemplateName).
+			WithField("pod", pod.Name).Error(message)
```
Comment (alexec) [inline workflow/controller/operator.go:1097]: warning

**Pull request merged**